common/v2/features/Dashboard/components/WalletBreakdown.tsx

```diff
@@ -103,3 +115,21 @@ export default function WalletBreakdown() {
     </div>
   );
 }
+
+function mapContextToProps(WrappedComponent: any) {
+  const WithContext = (props: any) => (
+    <AccountContext.Consumer>
+      {accountContext => (
+        <CurrentsContext.Consumer>
+          {currentsContext => (
+            <WrappedComponent {...props} {...accountContext} {...currentsContext} />
+          )}
+        </CurrentsContext.Consumer>
+      )}
+    </AccountContext.Consumer>
+  );
+
+  return WithContext;
+}
```

Comment:
I'd rather use the existing render prop components `Context.Provider` and `Context.Provider`. Render props and hooks are considered to be more flexible than HoCs, and HoCs are also very difficult to use with TypeScript. If we had to keep this HoC for some reason, we should remove all `any` types and use generic parameters (though this would be easier to fix with render props).

Reply:
Uodated to use useContext. I believe the new version remove all the hassles.
Long live Hooks!